substra/sdk/backends/remote/backend.py

```diff
@@ -127,15 +127,16 @@ def add(self, spec, exist_ok=False, spec_options=None):
 
         with spec.build_request_kwargs(**spec_options) as (data, files):
             response = self._add(asset_type, data, files=files, exist_ok=exist_ok)
-        # The backend has inconsistent API responses when getting or adding an asset
-        # (with much less data when responding to adds).
-        # A second GET request hides the discrepancies.
-        # Do not do this with a compute plan or we lose the id_to_key field
+        # The backend returns only the key, except for the compute plan: it returns the
+        # whole object (otherwise we lose the id_to_key field)
         if asset_type == schemas.Type.ComputePlan:
             return response
 
-        key = _get_asset_key(response)
-        return self.get(asset_type, key)
+        if get_asset:
+            key = _get_asset_key(response)
```

Comment:
This is the only use of _get_asset_key and here it can never be a compute plan (handled a couple of lines above). You could probably remove the function and just do `return self.get(asset_type, reponse['key']`